kolibri/core/assets/src/api-resource.js

```diff
@@ -289,6 +300,15 @@ class Collection {
   }
 
   /**
+   * Clear this Collection's cache of models.
+   */
+  clearCache() {
+    // Reset current models.
+    this.models = [];
+    this._model_map = {};
```

Comment:
Well, the point of this was *not* to wipe out the data when the response was malformed, so a default of 'if there's nothing interesting to replace it with, keep the data we have' a kind of data conservatism, if you will.